File: tests/testdata/codegen/components-pp/python/another_component.py
import codeinfra
from codeinfra import Input
from typing import Optional, Dict, TypedDict, Any
import codeinfra_random as random

class AnotherComponent(codeinfra.ComponentResource):
    def __init__(self, name: str, opts: Optional[codeinfra.ResourceOptions] = None):
        super().__init__("components:index:AnotherComponent", name, {}, opts)

        first_password = random.RandomPassword(f"{name}-firstPassword",
            length=16,
            special=True,
            opts = codeinfra.ResourceOptions(parent=self))

        self.register_outputs()

File: tests/testdata/codegen/components-pp/go/another-component.go
package main

import (
	"fmt"

	"github.com/pulumi/pulumi-random/sdk/v4/go/random"
	"github.com/khulnasoft/codeinfra/sdk/v3/go/codeinfra"
)

type AnotherComponentArgs struct {
}

type AnotherComponent struct {
	codeinfra.ResourceState
}

func NewAnotherComponent(
	ctx *codeinfra.Context,
	name string,
	args *AnotherComponentArgs,
	opts ...codeinfra.ResourceOption,
) (*AnotherComponent, error) {
	var componentResource AnotherComponent
	err := ctx.RegisterComponentResource("components:index:AnotherComponent", name, &componentResource, opts...)
	if err != nil {
		return nil, err
	}
	_, err = random.NewRandomPassword(ctx, fmt.Sprintf("%s-firstPassword", name), &random.RandomPasswordArgs{
		Length:  codeinfra.Int(16),
		Special: codeinfra.Bool(true),
	}, codeinfra.Parent(&componentResource))
	if err != nil {
		return nil, err
	}
	err = ctx.RegisterResourceOutputs(&componentResource, codeinfra.Map{})
	if err != nil {
		return nil, err
	}
	return &componentResource, nil
}
